specfile/specfile.py

```diff
@@ -50,9 +68,13 @@ class Specfile:
         autosave: Whether to automatically save any changes made.
     """
 
+    ENCODING_ARGS: EncodingArgs = {"encoding": "utf8", "errors": "surrogateescape"}
+
     def __init__(
         self,
-        path: Union[Path, str],
+        path: Optional[Union[Path, str]] = None,
+        file: Optional[IOBase] = None,
```

Comment:
I think this should be annotated with `typing.IO[str]` (a stream opened in string mode) and not `IOBase`.

Reply:
Or `typing.IO[str] | typing.IO[bytes]`, since the code also works with files opened in binary mode.